test/Sema/diag_constantness_check.swift

```diff
@@ -361,3 +361,21 @@ func testConstructorAnnotation(x: Int) {
   let _ = ConstructorTest(x)
     // expected-error@-1 {{argument must be an integer literal}}
 }
+
+// Test closure expressions
+
+func funcAcceptingClosure(_ x: () -> Void) {
+}
+
+func testCallsWithinClosures(s: String) {
+  funcAcceptingClosure {
+    constantArgumentFunction(s)
+    // expected-error@-1 {{argument must be a string literal}}
+  }
+  funcAcceptingClosure {
+    constantArgumentFunction(s)
+    // expected-error@-1 {{argument must be a string literal}}
+    constantArgumentFunction(s)
+    // expected-error@-1 {{argument must be a string literal}}
+  }
+}
```

Comment:
Can we add a test case that has nested single-expression closure. E.g. 
```swift
funcAcceptingClosure {
     funcAcceptingClosue { 
         constantArgumentFunction(s)
    }
}
``` 

```swift
funcAcceptingClosure {
     NormalFunctionCall() 
     funcAcceptingClosue { 
         constantArgumentFunction(s)
    }
}
```

Reply:
Another interesting test case, is using the the single expression closure with in a larger expression
```
funcAcceptingClosure {
   constantArgumentFunctionReturningIntCollection(s)
}
.filter(_ > 0) 
.map( _ + 1)
```
```
funcAcceptingClosure {
   constantArgumentFunctionReturningInt(s)
} + x * y
```

Reply:
One more suggestion is to add a non-trailing case e.g. `let _ = { constantArgumentFunctionReturningIntCollection(s) }`

Reply:
Also, some negative cases where diagnostics shouldn't appear: 
```
funcAcceptingClosure {
     constantArgumentFunction(1)
     constantArgumentFunction("some literal")
     constantArgumentFunction("string with a single interpolation \(x)")
   }
```
Can we also add similar test cases to `diag_constantness_check_os_log.swift`. You can replace `constantArgumentFunctions(x)` by `_osLogTestHelper("Minimum integer value: \(Int.min, format: formatOpt)")`  and for negative test cases where there should no diagnostics, replace `formatOpt` by something like .hex etc as in the examples in the test suite.